### PR TITLE
ComputeCpp ASP support

### PR DESCRIPTION
**Pull request by joeatodd**
Migrate from old version of ComputeCpp to Experimental AKA 'ASP'. This version offers full USM support which the simulation requires. In particular, the OpenCL backend for Intel GPU ('neo') doesn't work with the old version.

### CODE REVIEW

File: src_sycl/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,10 @@ if(BACKEND STREQUAL "COMPUTECPP")
 
 	set(CMAKE_CXX_COMPILER ${ComputeCpp_DEVICE_COMPILER_EXECUTABLE})
 
+	set_property(TARGET ${BINARY_NAME} APPEND
+        PROPERTY COMPILE_OPTIONS
+		-std=c++17 -sycl-std=2020)
```
Comment (DuncanMcBain) [inline src_sycl/CMakeLists.txt]: you can set the properties `CXX_STANDARD 17` and `CXX_STANDARD_REQUIRED` to add the correct C++ standard flags. I will soon add support for setting the SYCL standard in a similar fashion.

File: cmake/ComputeCppConfig.cmake [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #.rst:
-# FindComputeCpp
+# ComputeCppConfig
```
Comment (DuncanMcBain) [inline cmake/ComputeCppConfig.cmake]: I would recommend against adding this file to the repository. The idea is that ComputeCpp distributes the Config file, such that any user who downloads ComputeCpp can specify `-DComputeCpp_DIR=/path/to/computecpp/lib/cmake` (or some similar path) to use it directly from the package, without needing the code in the repository. (Besides, even if you do want to have this file in-repo, the one in the package is slightly out-of-date for the moment. The changes in this PR are effectively going *back* in time.)

Comment (joeatodd) [reply cmake/ComputeCppConfig.cmake]: Good point @DuncanMcBain - let me test that locally & make the change

File: src_sycl/CMakeLists.txt (hunk 1 above)
Comment (joeatodd) [reply src_sycl/CMakeLists.txt]: Done, as discussed offline.

**Pull request merged**